src/puptoo/process/profile.py

```diff
@@ -392,19 +395,30 @@ def system_profile(
             catch_error("uname", e)
             raise
 
-    if redhat_release:
+    if redhat_release and os_release:
         try:
+            # The 'redhat_release.rhel' may be "Red Hat Enterprise Linux xxx"
             profile["os_release"] = redhat_release.rhel
```

Comment:
So, are we populating `os_release` for other OS as well? I thought this block (i.e checking `redhat_release` is not null) is only for RHEL systems. Please correct me.

Reply:
I think so.
But here I just try my best to make sure that the new code stays compatible with the original code. 
The original code, in fact, didn't consider if the `redhat_release` is correct or not, and so does the new code.

From my perspective, I think we should update this line as 
```python
profile["os_release"] = os_release.release
```
which is a bit accurate and shows the `os_release` as `RHEL` OR `"Oracle", "Fedora"`, etc, when it's not an RHEL System. See [here](https://github.com/RedHatInsights/insights-core/blob/3.0/insights/combiners/os_release.py#L31)

Reply:
Confirmed. Yes, `operating_system` field can be for any OS - [inventory_schema](https://github.com/RedHatInsights/inventory-schemas/blob/master/tests/utils/valids.py#L80).
So does that mean, `redhat_release.rhel` (as per current code) can give us a release version of ANY OS? 
[.rhel](https://github.com/RedHatInsights/insights-core/blob/master/insights/parsers/redhat_release.py#L116) is not specific to RHEL?

Reply:
You are right. I made a mistake in the former comment, the `os_release` should not be `"RHEL"` OR `"Oracle", "Fedora"`, it should be the `release version of the OS`. Sorry about that.

The [redhat_release.rhel](https://github.com/RedHatInsights/insights-core/blob/master/insights/parsers/redhat_release.py#L116) tries to give the RHEL release version, just according to the literal text of the `/etc/redhat-release`, even when it isn't RHEL since the `/etc/redhat-release` can be mocked as RHEL freely (we do have such cases).

If here, we just want to show the release version ONLY for RHEL in this `os_release` field, we can do such change:
```python
if os_release.is_rhel:
   profile["os_release"] = redhat_release.rhel
else:
   profile["os_release"] = "null"   # or None
```
Does this work?